index.js

```diff
@@ -1422,6 +1434,90 @@ module.exports = class Autobase extends ReadyResource {
     this._rebooted()
   }
 
+  async _fork ({ indexers, length }) {
+    await this._applyState.finalize(this._applyState.system.core.key)
+
+    const store = this._viewStore.atomize()
+
+    const manifests = await store.getIndexerManifests(indexers)
+
+    const sysCore = store.get({ name: '_system' })
+    const encCore = store.get({ name: '_encryption' })
+
+    await sysCore.truncate(length)
+
+    const system = new SystemView(sysCore)
+    const encryption = new EncryptionView(this, encCore)
+
+    await system.ready()
+    await encryption.ready()
+
+    const entropy = system.getEntropy(indexers)
+
+    const manifestVersion = 2 // needed to signal new encryption
+
+    const views = []
+    for (const view of this._applyState.views) {
+      const v = this._applyState.getViewFromSystem(view, system)
+      const indexedLength = v ? v.length : 0
+
+      const session = store.get({ name: view.name })
+      await session.ready()
+
+      await session.truncate(indexedLength)
+
+      const manifestData = session.manifest.version <= 1 ? c.encode(messages.ManifestData, { encryption: { legacyBlocks: indexedLength } }) : null
+      const ref = await this._migrateView(manifests, session, view.name, session.length, manifestVersion, entropy, null, manifestData)
+
+      // update key
+      views.push({ key: ref.core.key, length: indexedLength })
+    }
+
+    // internal views are explicitly migrated
+
+    await encryption.update(entropy)
+    const encRef = await this._migrateView(manifests, encCore, '_encryption', encCore.length, manifestVersion, entropy, null, null)
+
+    const sysEncryption = encryption.getSystemEncryption({ forceNonCompat: sysCore.length })
+
+    await system.fork(indexers, manifests, views, sysEncryption)
+    const manifestData = system.core.manifest.version <= 1 ? c.encode(messages.ManifestData, { encryption: { legacyBlocks: length } }) : null
+
+    const ref = await this._migrateView(manifests, system.core, '_system', length, manifestVersion, entropy, [encRef.core.key], manifestData)
+
+    const local = store.getLocal()
+    await local.ready()
+
+    const pointer = c.encode(messages.BootRecord, {
+      key: ref.core.key,
+      internalViews: [sysCore.length, encCore.length],
```

Comment:
remind me why we need to track these now? i can't remember but it was due to truncatinos right?